lib/src/network/header_proxy_protocol.rs

```diff
@@ -0,0 +1,166 @@
+use std::net::{SocketAddr, SocketAddrV4, SocketAddrV6};
```

Comment:
this file should probably be in `lib/src/network/protocol/`, since it is only linked to the protocol implementation. Maybe make the following moves:

- `lib/src/network/header_proxy_protocol.rs` -> `lib/src/network/protocol/proxy_protocol/header.rs`
- `lib/src/network/protocol/proxy_protocol.rs` -> `lib/src/network/protocol/proxy_protocol/mod.rs`

Reply:
Done

Reply:
Done